src/ba.py

```diff
@@ -36,6 +36,12 @@ def strict_border_array(x: str) -> list[int]:
     for i in range(len(x)-1):
         if (ba[i] != 0 and x[ba[i]] != x[i+1]):
             bax[i] = ba[i]
+        #Search for shorter border
+        elif(ba[i] != 0 and x[ba[i]] == x[i+1]):
+            for j in range(1,ba[i]):
+                if (x[ba[i]-j] != x[i+1] and x[0:ba[i]-j]==x[i-ba[i]+j+1: i+1]):
```

Comment:
Have a think of what the complexity of this might be. How long does it take to slice a string `x[i:j]`?